### PR TITLE
SOC-878 Adding active state to birthdate field

### PR DESCRIPTION
**Pull request by bkoval**
@kvas-damian @rafalkalinski https://wikia-inc.atlassian.net/browse/SOC-878

This ticket was filed to fix the state on a placeholding input type=date, but I added an active state on the new date field, so the PR is targeted at SOC-480.

### CODE REVIEW

File: front/scripts/auth/common/BirthdateInput.ts [hunk 1]
```diff
@@ -34,13 +34,23 @@ class BirthdateInput {
...
+		var firstInput: HTMLInputElement = <HTMLInputElement>this.fakeInputs[0],
+			inputContainer = <HTMLElement> this.wrapper.parentElement,
+			target: HTMLElement;
 
-		this.realInput.addEventListener('focus', (() => {
-			this.realInput.type = 'hidden';
-			this.wrapper.classList.remove('hide');
-			firstInput.focus();
-		}).bind(this));
+		inputContainer.addEventListener('focus', (() => {
```
Comment (rogatty) [inline front/scripts/auth/common/BirthdateInput.ts]: Missing return type :P

File: front/scripts/auth/common/BirthdateInput.ts [hunk 2]
```diff
@@ -34,13 +34,23 @@ class BirthdateInput {
...
+			target = <HTMLElement> event.target;
+			if (target === this.realInput) {
+				this.realInput.type = 'hidden';
+				this.wrapper.classList.remove('hide');
+				firstInput.focus();
+			}
+			this.wrapper.classList.add('focused');
+		}).bind(this), true);
+
+		inputContainer.addEventListener('blur', (() => {
```
Comment (rogatty) [inline front/scripts/auth/common/BirthdateInput.ts]: Missing return type.

**Comment by rafalkalinski** [comment]
except missing return types, lgtm :)

**Comment by kvas-damian** [comment]
I see a lot is happening here, can't we just update styles in: https://github.com/Wikia/mercury/blob/dev/front/styles/auth/module/_forms.scss#L48 to reflect styleguide animations?
Or is that because we're replacing input with 3 different?

**Comment by bkoval** [comment]
The second thing. Input[type=date] will be replaced with three number inputs, that's why we need to apply these styles to their wrapper which acts like a one big input consisting of three smaller ones.

**Comment by kvas-damian** [comment]
LGTM

**Pull request merged**